### PR TITLE
Building with forms

### PR DESCRIPTION
**Pull request by AdfPizarro**
Setup the DB and migration for the users
Included form to create and edit user

Add validators on the server-side
Display red color on the fields with validation errors

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    # @user = User.new(whitelisted_user_params)
+    if @user.save
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def new
+    @user = User.new(email: 'example@example.com')
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: Please, move `example@example.com` in the placeholder of the form

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  def update
+    @user = User.find(params[:id])
+    @user.update(user_params)
+  end
```
Comment (maelfosso) [inline app/controllers/users_controller.rb:22]: It's not working sorry.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+class UsersController < ApplicationController
+  def create
+    @user = User.new(user_params)
+    # @user = User.new(whitelisted_user_params)
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: Remove comments

File: config/routes.rb [hunk 4]
```diff
@@ -1,3 +1,3 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :users, only: %i[create new edit index update]
```
Comment (maelfosso) [inline config/routes.rb:2]: As indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png), the routes are restricted.

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+     <%= f.text_field :username %>
+
+    <%= f.label :email %><br />
+     <%= f.text_field :email %>
+  
+     <%= f.label :password %><br />
+     <%= f.password_field :password %>
+
+     <%= f.submit %>
+ <% end %> -->
```
Comment (maelfosso) [inline app/views/users/new.html.erb:23]: The first form implementation is not conform to the requirements.
Take a look of the requirements [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png)

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+<% end %>
+
+<%= form_for(@user) do |f| %>
+   <div>
+     <%= f.label :username, 'Introduce your username' %><br />
+     <% if @user.errors[:username] %>
+	<%= f.text_field :username, placeholder: "Your username", :class=> "error" %>
+     <% else %>
+ 	<%= f.text_field :username, placeholder: "Your username" %>
+     <% end %>
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: Even it's a good idea, the implementation is not good. 
You must avoid logic in views. Please, use instead view helpers for that.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello 
Thank you for the job done so far.
However there are some changes to do. Read them in the comments below.

Once the code updated, kindly submit a new code review request. 

Thanks
Mael FOSSO

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (AdfPizarro) [reply app/controllers/users_controller.rb]: Line removed

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (AdfPizarro) [reply app/controllers/users_controller.rb]: Moved to the view

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (AdfPizarro) [reply app/controllers/users_controller.rb:22]: Created view to list all the users and links to edit each one

File: config/routes.rb (hunk 4 above)
Comment (AdfPizarro) [reply config/routes.rb:2]: We nee the routes in order to be able to update and list users, this instruction doesn't make sense

File: app/views/users/new.html.erb (hunk 5 above)
Comment (AdfPizarro) [reply app/views/users/new.html.erb:23]: Added the original form in plain html, modified to send and object instead

File: app/views/users/new.html.erb (hunk 6 above)
Comment (AdfPizarro) [reply app/views/users/new.html.erb]: Removed unnecessary logic from the views added method to list errors

**Review by jofer86 — APPROVED** [review]
## :wave: :wave: Hello! I'm Jorge, and  I'm here to review this Pull Request, let's take a look shall we :eye: :eye: :eye: 
- Changes marked as **[OPTIONAL]** are not hard requirements but are nice to have.
- All of the required changes were met, well done!! This project is vanquished!!
## `const PROJECT_STATUS = 'Approved'` 🚀 🚀 🚀
--------------------------------------------------------------------
Cheers, 
Jorge Fernando Rincón Figueroa.
Technical Support Engineer.
**Slack: @jofer86**

**Pull request merged**